cmd/api/src/database/saved_queries.go

```diff
@@ -74,3 +84,20 @@ func (s *BloodhoundDB) SavedQueryBelongsToUser(ctx context.Context, userID uuid.
 		return false, nil
 	}
 }
+
+// GetSavedQueriesSharedWithUser returns all the saved queries that the given userID has access to, including global queries
+func (s *BloodhoundDB) GetSavedQueriesSharedWithUser(ctx context.Context, userID int64) (model.SavedQueries, error) {
+	savedQueries := model.SavedQueries{}
+
+	result := s.db.WithContext(ctx).Where("shared_to_user_id = ?", userID).Find(&savedQueries)
+
+	return savedQueries, CheckError(result)
+}
+
```

Comment:
Moved these methods here. They pertain to the permissions but return a SavedQuery model, which made sense to put them here to me.
This also gives power to the end user implementing the endpoints something akin to:

```go
sharedQueries := GetSavedQueriesSharedWithUser(ctx, id)
globalQueries := GetSavedQueriesSharedGlobally(ctx)

return SavedQueriesSharedWithUser {
    GlobalQueries: globalQueries,
    SharedQueries: sharedQuereies
}
```